plugins/modules/zos_operator.py

```diff
@@ -256,123 +274,41 @@ def parse_params(params):
 
 
 def run_operator_command(params):
-    # Usage: (rexfile) wait_time_s command [-v] [-n]
-    #       -v: print out verbose security information
-    #       -n: nowait
-
-    script = """/*rexx*/
-wait_time = __argv.2
-command = __argv.3
-verbosemode = __argv.4
-nowait = __argv.5
-
-Address 'TSO'
-IsfRC = isfcalls( "ON" )
-showoutput = 0
-waitmsg = " WAIT"
-if nowait == "NOWAIT" then do
-  waitmsg = ""
-  end
-
-verbose = ""
-if verbosemode == "VERB" then do
-  showoutput = 1
-  ISFSECTRACE="ON"
-  verbose = "( VERBOSE" waitmsg ")"
-  end
-else do
-  if waitmsg == " WAIT" then do
-    verbose = "( WAIT )"
-    end
-  end
-
-if wait_time > 0 then do
-  ISFDELAY=wait_time
-  end
-
-sdsfcmd = False
-
-fwd = WORD(command, 1)
-ffwd = translate(fwd)
-fch = SUBSTR( fwd, 1, 1)
-if( ffwd == 'QUERY' | ffwd == 'SET' | ffwd == 'WHO' | fch == '/' ) then
-  do
-    sdsfcmd = True
-  end
-if sdsfcmd == True then
-  do
-    address SDSF "ISFEXEC " command verbose
-  end
-else
-  do
-    address SDSF "ISFEXEC '/"command"'" verbose
-  end
-saverc = rc
-
-IsfRC = isfcalls( "OFF" )
-trace Off
-if isfulog.0 > 0 then
-  do
-    do ix=1 to isfulog.0
-      say isfulog.ix
-    end
-  end
-if isfresp.0 > 0 then
-  do
-    do ix=1 to isfresp.0
-      say isfresp.ix
-    end
-  end
-if showoutput > 0 then
-  do
-    SAY "===================="
-    SAY "result code: " saverc
-    SAY "===================="
-
-    say ""
-    say "Action messages"
-    say isfmsg
-    say ""
-    do ix=1 to isfmsg2.0
-      say isfmsg2.ix
-    end
-    say ""
-  end
-EXIT saverc
-"""
     module = AnsibleModuleHelper(argument_spec={})
 
-    fulline = " " + str(params.get("wait_time_s")) + " "
-
-    fulline += '"' + params.get("cmd") + '"'
+    kwargs = {}
 
     if params.get("verbose"):
-        fulline += " VERB"
-    else:
-        fulline += " QUIET"
+        kwargs.update({"verbose": "verbose"})
+
+    if params.get("debug"):
+        kwargs.update({"debug": "debug"})
 
     if params.get("wait"):
-        fulline += " WAIT"
-    else:
-        fulline += " NOWAIT"
+        wait = params.get("wait_time_s")
+        if wait:
+            kwargs.update({"timeout": "{0}".format(wait)})
+            kwargs.update({"parameters": "ISFDELAY={0}".format(wait)})
```

Comment:
I glanced at ZOAU's [opercmd command api](https://www.ibm.com/docs/en/zoau/1.2.0?topic=modules-opercmd) and so no support the timeout and ISFDELAY which in this case both will be set to either True or False depending on the users entry. These are not returned parameters from what I gather from the doc so i wonder why we kept them.

Reply:
I reach out to ZOAU and also went digging through the code and could not find the `timeout` support we requested, for now ill approve and merge and open an issue that would require us to come back and revisit this issue. @richp405